### PR TITLE
Even more validate opt

### PR DESCRIPTION
**Pull request by matthewrmshin**
~Graph parser: improve results data structure. Unify variable names where possible.~

Conditional simplifier:
* faster `listify` method, convert expression directly into nested list.
* ~new `deep_replace` method~

`get_graph_raw`:
* More caching.
* ~Reduce number of `(name, point)` <=> `name.point` conversions, but it means there is an API change.~

graphing:
* Don't style node and edge on validation.

Initial tests suggest a good reduction of memory usage (from 920MB to 750MB) and a small reduction of user time and elapsed time (about 6%) to validate the nasty suite in my work environment.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Profiling results (cylc validate) are a little mixed and somewhat hard to interpret:

__Key__

```
-   Small but measurable decrease
--  Decrease
++  Increase
=   No change
~   Mixed - worse scaling
```
__Results__

| Experiment | Memory | CPU / Elapsed Time |
| --- | --- | --- |
| complex | -- | = |
| hello-world | -- | = |
| diamond | = |  ~ |
| family-trigger | - | - |
| busy | = | -- |
| lazy | = | ++ |

_Lazy_
elapsed-time

_Diamond_
elapsed-time

**Comment by matthewrmshin** [comment]
I'll have a go at breaking up the change into chunks.

**Comment by matthewrmshin** [comment]
(I have replaced the branch with a new and smaller set of changes that should be easier to understand. The smaller change seems to achieve almost similar reduction in CPU and memory on validation of the nasty horror suite.)

File: lib/cylc/config.py [hunk 1]
```diff
@@ -1688,17 +1688,10 @@ def get_graph_raw(self, start_point_string, stop_point_string,
...
-                if i in members[j]:
-                    if i in clf:
-                        clf.remove(i)
-
         clf_map = {}
-        for family_name in clf:
-            clf_map[family_name] = (
-                self.runtime['first-parent descendants'][family_name])
+        for name in self.closed_families:
+            if any([name not in members[i] for i in self.closed_families]):
```
Comment (oliver-sanders) [inline lib/cylc/config.py]: This if statement only evaluates once the entire list has been constructed, switch to a generator to accelerate this https://github.com/metomi/rose/pull/2064#discussion_r112664279.

**Comment by matthewrmshin** [comment]
I think I am done.

**Comment by matthewrmshin** [comment]
Nasty suite resource usage in my environment:

Resource | master | branch
------------|--------|-------
User time   | 190s   | 151s
Elapsed time| 3:11   | 2:34
Max memory  | 895MB  | 722MB

**Comment by matthewrmshin** [comment]
Version               | Run          | Elapsed Time (s)| CPU Time - Total (s)| Max Memory (kb)
----------------------|--------------|-----------------|---------------------|----------------
master                | complex suite| 8.1             | 8.0                 | 71424.0        
branch| complex suite| 6.3             | 6.2                 | 70120.0        
master                | diamond 1000 | 6.2             | 6.2                 | 101772.0       
branch| diamond 1000 | 4.7             | 4.7                 | 100400.0

**Pull request merged**